lib/synapse/service_watcher/zookeeper_poll.rb

```diff
@@ -0,0 +1,77 @@
+require 'synapse/service_watcher/base'
+require 'synapse/service_watcher/zookeeper'
+
+require 'zk'
+require 'thread'
+
+class Synapse::ServiceWatcher
+  class ZookeeperPollWatcher < ZookeeperWatcher
+    def start
+      log.info 'synapse: ZookeeperPollWatcher starting'
+
+      @should_exit = false
+      @thread = nil
+      @poll_interval = @discovery['polling_interval_sec']
+
+      zk_connect do
+        @thread = Thread.new {
+          log.info 'synapse: zookeeper polling thread started'
+
+          # last_run is shifted by a random jitter in order to spread the first
+          # discover call of multiple Synapses. This helps to spread load.
+          # As long as the beginning is spread, the future discovers will also
+          # be spread.
+          last_run = Time.now - rand(@poll_interval)
+
+          until @should_exit
+            now = Time.now
+            elapsed = now - last_run
+
+            if elapsed >= @poll_interval
```

Comment:
what happens if polling_interval_sec is not set? default value?

Reply:
I was wary of setting a default value because I'm not sure what a good default would be--the acceptable load depends on how many Synapses you have running.

Reply:
Just want to make sure, if the flag is not set, synapse shouldn't crash.

The default value can be any value sane and safe, in case some one put method as `zookeeper_poll` but doesn't set `polling_interval_sec`.